File: AkkaSupervisorExample/src/main/java/org/akka/essentials/supervisor/example3/WorkerActor.java
package org.akka.essentials.supervisor.example3;

import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class WorkerActor extends UntypedActor {
	LoggingAdapter log = Logging.getLogger(getContext().system(), this);
	@SuppressWarnings("unused")
	private int state = 0;

	@Override
	public void preStart() {
		log.info("Starting WorkerActor instance hashcode #" + this.hashCode());
	}

	public void onReceive(Object o) throws Exception {
		if (o instanceof Integer) {
			Integer value = (Integer) o;
			state = value;
			log.info("Received a message " + value);
		} else if (o instanceof Result) {
			getSender().tell(state);
		} else {
			throw new IllegalArgumentException("Wrong Arguement");
		}
	}

	@Override
	public void postStop() {
		log.info("Stopping WorkerActor instance hashcode #" + this.hashCode());

	}
}

File: AkkaSupervisorExample/src/main/scala/org/akka/essentials/supervisor/example3/WorkerActor.scala
package org.akka.essentials.supervisor.example3
import akka.actor.actorRef2Scala
import akka.actor.Actor
import akka.actor.ActorLogging

class WorkerActor extends Actor with ActorLogging {
  import org.akka.essentials.supervisor.example1.Result
  var state: Int = 0;

  override def preStart() {
    log.info("Starting WorkerActor instance hashcode #" + this.hashCode());
  }
  override def postStop() {
    log.info("Stopping WorkerActor instance hashcode #" + this.hashCode());
  }
  def receive: Receive = {
    case value: Int =>
        state = value;
    case result: Result =>
      sender ! state
    case _ =>
      context.stop(self)
  }
}